src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -101,20 +101,46 @@ static string DecodeFrom64(string encodedData)
         /// <returns>Subscription addresses that were found for the provided messageTypes</returns>
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var subscribers = new List<Subscriber>();
+            var subscribers = new HashSet<Subscriber>(SubscriberComparer.Instance);
             var table = client.GetTableReference(subscriptionTableName);
 
             foreach (var messageType in messageTypes)
             {
-                var query = new TableQuery<Subscription>().Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, messageType.ToString()));
+                var name = messageType.TypeName;
+                var lowerBound = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, name);
+                var upperBound = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThan, GetUpperBound(name));
+                var query = new TableQuery<Subscription>().Where(TableQuery.CombineFilters(lowerBound, "and", upperBound));
```

Comment:
remark: there is a potential table scan here, but as this table is usually small I don't think it is an issue. As ranges are kept close together I can see this potentially hit a few storage nodes, but never all